openedx/core/djangoapps/user_authn/api/form_fields.py

```diff
@@ -163,3 +211,129 @@ def add_job_title_field():
+        'error_message': accounts.REQUIRED_FIELD_LAST_NAME_MSG,
+    }
+
+
+def add_mailing_address_field():
+    """
+    Returns the mailing address field description
+    """
+    # Translators: This label appears above a field
+    # meant to hold the user's mailing address.
+    mailing_address_label = _("Mailing address")
+
+    return {
+        'name': 'mailing_address',
+        'type': SUPPORTED_FIELDS_TYPES['TEXTAREA'],
+        'label': mailing_address_label,
+        'error_message': accounts.REQUIRED_FIELD_MAILING_ADDRESS_MSG,
+    }
+
+
+def add_state_field():
+    """
+    Returns a State/Province/Region field to a description
+    """
+    # Translators: This label appears above a field
+    # which allows the user to input the State/Province/Region in which they live.
+    state_label = _("State/Province/Region")
+
+    return {
+        'name': 'state',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': state_label,
+        'error_message': accounts.REQUIRED_FIELD_STATE_MSG,
+    }
+
+
+def add_city_field():
+    """
+    Returns a city field to a description
+    """
+    # Translators: This label appears above a field
+    # which allows the user to input the city in which they live.
+    city_label = _("City")
+
+    return {
+        'name': 'city',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': city_label,
+        'error_message': accounts.REQUIRED_FIELD_CITY_MSG,
+    }
+
+
+def add_honor_code_field(separate_honor_and_tos=False):
+    """
+    Returns a honor code field to a description
+    """
+
+    # Separate terms of service and honor code checkboxes
+    if separate_honor_and_tos:
+        terms_label = "Honor Code"
+        field_type = 'CHECKBOX'
+
+    # Combine terms of service and honor code checkboxes
+    else:
+        # Translators: This is a legal document users must agree to
+        # in order to register a new account.
+        terms_label = "Terms of Service and Honor Code"
+        field_type = 'TEXT'
+
+    # Translators: "Terms of Service" is a legal document users must agree to
+    # in order to register a new account.
+    platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+    error_msg = f'You must agree to the {platform_name} {terms_label}'
+    return {
+        'name': terms_label,
+        'type': SUPPORTED_FIELDS_TYPES[field_type],
+        'error_message': error_msg,
+    }
+
+
+def add_country_field():
+    """
+    Returns a country name to a description
+    """
+    # Translators: This label appears above a dropdown menu on the registration
+    # form used to select the country in which the user lives.
+
+    country_label = _("Country or Region of Residence")
+
+    return {
+        'name': 'country',
+        'type': SUPPORTED_FIELDS_TYPES['SELECT'],
+        'label': country_label,
+        'error_message': accounts.REQUIRED_FIELD_COUNTRY_MSG,
+        'options': list(countries),
+    }
```

Comment:
we are displaying country field directly on frontend, we don't need to send all these extra fields.